pkg/controller/datavolume-controller.go

```diff
@@ -1666,7 +1680,7 @@ func (r *DatavolumeReconciler) validateAdvancedCloneSizeCompatible(
 	if err != nil {
 		return false, err
 	}
-	if usableSpace.Cmp(targetRequest) > 0 {
+	if usableSpace.Cmp(targetRequest) > 0 && !targetRequest.IsZero() {
```

Comment:
like I asked in a previous commit isn't it ok to also have !hasTargetRequest in case of storage API?